### PR TITLE
fix: update props for `loadingSpinner` to use `isSubmitting`

### LINKED ISSUE #362: Update LoadingSpinner to Use isSubmitting from react-hook-form
## Description
Currently, the `LoadingSpinner` component is conditionally rendered based on a useState `isLoading` prop. We need to update this implementation to leverage the isSubmitting state provided by react-hook-form, ensuring the spinner reflects the form's actual submission state.

## Criteria
- [x] Use `{ formState: isSubmitting }` from `useForm()`
- [x] Replace `{isLoading && <LoadingSpinner isLoading={true} />}` with `{<LoadingSpinner isLoading={isSubmitting} />}`.
- [x] Ensure that the isSubmitting state from react-hook-form is correctly passed to the LoadingSpinner component.
- [x] Test the component to ensure the spinner displays correctly during form submission.

### PR DESCRIPTION
**Pull request by Akikaze1119**
## Overview
I changed `isLoading` props in `LoadingSpinner` component to use `isSubmitting` state from `react-hook-form`.

## Changes
`src/components/accordion-assignment-sheet/AccordionAssignmentSheet.tsx`
`src/components/sharehouses-management/RotationCycles.tsx`
-  Changed `isLoading` props in `LoadingSpinner` component to pass `isSubmitting` state as a prop.

`src/components/ui/drawers/auth/LogOutDrawer.tsx`
- Updated to use `CommonDrawer` component.
- Modified a button label ('Log Out' ⇨ 'Log out')

## Review points
- [x] Replace `{isLoading && <LoadingSpinner isLoading={true} />}` with `{<LoadingSpinner isLoading={isSubmitting} />}`.
- [x] Ensure that the isSubmitting state from react-hook-form is correctly passed to the LoadingSpinner component.
- [x] Test the component to ensure the spinner displays correctly during form submission.
- [x] Check all files using `LoadingSpinner` to ensure no instances are missed.

## Notes
`/src/components/ui/drawers/auth/EmailVerificationDrawer.tsx`
This file is not subject to modification because it cannot use `isSubmitting` from react-hook-form.

## Assignee Checklist:

- [x] The base branch is correct (no accidental merges)
- [x] The branch name follows our branch naming rules
- [x] The PR title follows our PR title rules
- [x] My code follows our coding style
- [x] I have commented on my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings or errors in the console

## Reviewer Checklist:

- [ ] Code readability and simplicity
- [ ] Follows best practices and coding standards
- [ ] Understandable and maintainable code

### CODE REVIEW

**Review by HrRn — CHANGES_REQUESTED** [review]
Thank you for your work! I left comments so please check it out :)

File: src/components/accordion-assignment-sheet/AccordionAssignmentSheet.tsx [hunk 1]
```diff
@@ -150,7 +150,7 @@ export const AccordionAssignmentSheet = ({
 
   return (
     <>
-      {isSubmitting && <LoadingSpinner isLoading={true} />}
```
Comment (HrRn) [inline src/components/accordion-assignment-sheet/AccordionAssignmentSheet.tsx:153]: Could you please delete the curly brackets?

File: src/components/sharehouses-management/RotationCycles.tsx [hunk 2]
```diff
@@ -80,7 +80,7 @@ export const RotationCycles = ({
 
   return (
     <>
-      {isSubmitting ? <LoadingSpinner isLoading={true} /> : ''}
+      {<LoadingSpinner isLoading={isSubmitting} />}
```
Comment (HrRn) [inline src/components/sharehouses-management/RotationCycles.tsx]: Could you please delete the curly brackets?

File: src/components/accordion-assignment-sheet/AccordionAssignmentSheet.tsx [hunk 3]
```diff
@@ -150,7 +150,7 @@ export const AccordionAssignmentSheet = ({
...
+      {<LoadingSpinner isLoading={isSubmitting} />}
```
Comment (HrRn) [inline src/components/accordion-assignment-sheet/AccordionAssignmentSheet.tsx]: Could you please delete the curly brackets?

File: src/components/sharehouses-management/RotationCycles.tsx (hunk 2 above)
Comment (Akikaze1119) [reply src/components/sharehouses-management/RotationCycles.tsx]: @HrRn 
Thank you for your review!
Sorry, I deleted curly brackets.
`src/components/accordion-assignment-sheet/AccordionAssignmentSheet.tsx`
`src/components/sharehouses-management/RotationCycles.tsx`

Also, I found this file had the same isuue. Could you review this one too?
`src/components/login/Form.tsx`

**Review by HrRn — APPROVED** [review]
Thank you for your great work!

**Pull request merged**